pkg/integrations/mysqld_exporter/config.go

```diff
@@ -0,0 +1,177 @@
+	PerfSchemaFileInstancesFilter        string `yaml:"perf_schema_file_instances_filter"`
+	PerfSchemaFileInstancesRemovePrefix  string `yaml:"perf_schema_file_instances_remove_prefix"`
+	HeartbeatDatabase                    string `yaml:"heartbeat_database"`
+	HeartbeatTable                       string `yaml:"heartbeat_table"`
+	HeartbeatUTC                         bool   `yaml:"heartbeat_utc"`
+	MySQLUserPrivileges                  bool   `yaml:"mysql_user_privileges"`
+}
+
+// UnmarshalYAML implements yaml.Unmarshaler for Config.
+func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	*c = DefaultConfig
+
+	type plain Config
+	return unmarshal((*plain)(c))
+}
+
+// GetScrapers returns the set of *enabled* scrapers from the config.
+// Configurable scrapers will have their configuration filled out matching the
+// Config's settings.
+func GetScrapers(c Config) []collector.Scraper {
+	scrapers := map[collector.Scraper]bool{
+		&collector.ScrapeAutoIncrementColumns{}:                false,
+		&collector.ScrapeBinlogSize{}:                          false,
+		&collector.ScrapeClientStat{}:                          false,
+		&collector.ScrapeEngineInnodbStatus{}:                  false,
+		&collector.ScrapeEngineTokudbStatus{}:                  false,
+		&collector.ScrapeGlobalStatus{}:                        true,
+		&collector.ScrapeGlobalVariables{}:                     true,
+		&collector.ScrapeInfoSchemaInnodbTablespaces{}:         false,
+		&collector.ScrapeInnodbCmpMem{}:                        true,
+		&collector.ScrapeInnodbCmp{}:                           true,
+		&collector.ScrapeInnodbMetrics{}:                       false,
+		&collector.ScrapePerfEventsStatementsSum{}:             false,
+		&collector.ScrapePerfEventsWaits{}:                     false,
+		&collector.ScrapePerfFileEvents{}:                      false,
+		&collector.ScrapePerfIndexIOWaits{}:                    false,
+		&collector.ScrapePerfReplicationApplierStatsByWorker{}: false,
+		&collector.ScrapePerfReplicationGroupMemberStats{}:     false,
+		&collector.ScrapePerfReplicationGroupMembers{}:         false,
+		&collector.ScrapePerfTableIOWaits{}:                    false,
+		&collector.ScrapePerfTableLockWaits{}:                  false,
+		&collector.ScrapeQueryResponseTime{}:                   true,
+		&collector.ScrapeReplicaHost{}:                         false,
+		&collector.ScrapeSchemaStat{}:                          false,
+		&collector.ScrapeSlaveHosts{}:                          false,
+		&collector.ScrapeSlaveStatus{}:                         true,
+		&collector.ScrapeTableStat{}:                           false,
+		&collector.ScrapeUserStat{}:                            false,
+
+		// Collectors that have configuration
+		&collector.ScrapeHeartbeat{
+			Database: c.HeartbeatDatabase,
+			Table:    c.HeartbeatTable,
+			UTC:      c.HeartbeatUTC,
+		}: false,
```

Comment:
Please excuse my ignorance but would it but would it be possible to unmarschal `ScrapeHearbeat` directly from the YAML and thus avoid part of the extra config struct?

Reply:
It would be possible if we added yaml tags to the exporter repo (so you could configure it as `database` vs `Database`), but I don't think that change would be likely to get accepted.

Reply:
Ah, thanks for the explanation. :)